src/Transform/ONNX/Decompose.cpp

```diff
@@ -270,40 +271,46 @@ Value reverseWeightTensor(
 }
 
 // Calculate padding size used in Conv op from pads for ConvTranspose op.
-ArrayAttr getPadsConvTranspose(PatternRewriter &rewriter, Location loc,
-    ArrayAttr kernel, ArrayAttr pads, ArrayAttr dilation) {
-  // Calculate pads in generated Conv op by rewriting ConvTranspose op.
+ArrayAttr getPadsConvTranspose(
+    PatternRewriter &rewriter, Location loc, ONNXConvTransposeOp op) {
+  // Calculate pads for generated Conv op.
   // new_pads = kernel -  pads - 1
   // Reference: Dumoulin, Vincent, and Francesco Visin. "A guide to convolution
   // arithmetic for deep learning." arXiv preprint arXiv:1603.07285 (2016).
+  ONNXConvTransposeOpShapeHelper shapeHelper(op.getOperation(), {});
+  LogicalResult shapecomputed = shapeHelper.computeShape();
+  assert(succeeded(shapecomputed) &&
+         "unexpected inferShapes failuer for ConvTrans op");
+  SmallVector<IndexExpr, 2> kernelShape = shapeHelper.kernelShape;
+  SmallVector<int64_t, 2> dilations = shapeHelper.dilations;
+  DimsExpr pads = shapeHelper.pads;
+  assert(IndexExpr::isLiteral(kernelShape) && IndexExpr::isLiteral(pads) &&
+         "Currently only static dims are supported in spatial dims.");
+
   SmallVector<int64_t, 4> newPads;
   SmallVector<int64_t, 2> newKernel;
   // If `dilations` is not default [1, 1], `kernel` is updated by inserting
   // spaces in kernel elements.
   //   ex. kernel [2, 3] and dilation [2, 2], then new `kernel` is [3, 4]
-  for (unsigned int i = 0; i < kernel.size(); ++i)
+  for (unsigned int i = 0; i < kernelShape.size(); ++i)
     newKernel.emplace_back(
-        ArrayAttrIntVal(kernel, i) +
-        (ArrayAttrIntVal(kernel, i) - 1) * (ArrayAttrIntVal(dilation, i) - 1));
-
+        kernelShape[i].getLiteral() +
+        (kernelShape[i].getLiteral() - 1) * (dilations[i] - 1));
   // Calculate new pads. `kernel` size is doubled for the calculation.
-  for (unsigned int i = 0; i < kernel.size() * 2; ++i)
+  for (unsigned int i = 0; i < kernelShape.size() * 2; ++i)
     newPads.emplace_back(
-        newKernel[i % kernel.size()] - ArrayAttrIntVal(pads, i) - 1);
+        newKernel[i % kernelShape.size()] - pads[i].getLiteral() - 1);
   return rewriter.getI64ArrayAttr(newPads);
 }
 
 // Check if strides is unit strides.
-bool hasUnitStrides(ArrayAttr strides) {
-  SmallVector<int64_t, 3> vstrides;
-  for (unsigned int i = 0; i < strides.size(); ++i)
-    vstrides.emplace_back(ArrayAttrIntVal(strides, i));
-  return llvm::all_of(vstrides, [](int64_t s) { return s == 1; });
-}
-
-ArrayAttr createUnitStrides(PatternRewriter &rewriter, ArrayAttr strides) {
-  SmallVector<int64_t, 2> unitStrides(strides.size(), 1);
-  return rewriter.getI64ArrayAttr(unitStrides);
+bool hasUnitStrides(ONNXConvTransposeOp op) {
+  ONNXConvTransposeOpShapeHelper shapeHelper(op.getOperation(), {});
+  LogicalResult shapecomputed = shapeHelper.computeShape();
+  assert(succeeded(shapecomputed) &&
+         "unexpected inferShapes failuer for ConvTrans op");
```

Comment:
Please use `computeShapeAndAssertOnFailure` instead of `computeShape`, that includes assertion.

Reply:
Do we really need to call shapeHelper just for getting the stride attribute?

Reply:
Removed shapehelper in this function. Thanks.